app/mailers/building_operator_mailer.rb

```diff
@@ -0,0 +1,29 @@
+class BuildingOperatorMailer < ApplicationMailer
+  # Gets sent when a building operator is delegated questions but *hasn't been onboarded yet*
+  def new_user_delegated_email(user)
+    @user = user
+
+    # Manually generate a password reset link
+    raw, encrypted = Devise.token_generator.generate(BuildingOperator, :reset_password_token)
+    @user.reset_password_token = encrypted
+    @user.reset_password_sent_at = Time.now.utc
+    @user.save
```

Comment:
just curious, what is user? Is this a session or an actual new user object?

Reply:
It's the actual new user object, since the BuildingOperator model does `BuildingOperatorMailer.new_user_delegated_email(self)` when it's created

Reply:
so doesn't this function get called after the new user is created? Why save again?

Reply:
The encrypted password token has to get stored on the user for validation later (when the building operator actually clicks the link), so I'm setting it once the token gets generated here and then re-saving the user